### PR TITLE
1496 add failure

### PR DESCRIPTION
**Pull request by pythonissam**
Before submitting your PR, check that you've:

- [x] Bumped the version number
- [x] Documented new APIs with [Haddock markup](https://www.haskell.org/haddock/doc/html/index.html)
- [x] Added [`@since` declarations](http://haskell-haddock.readthedocs.io/en/latest/markup.html#since) to the Haddocks for new, public APIs

After submitting your PR:

- [x] Update the Changelog.md file with a link to your PR
- [ ] Check that CI passes (or if it fails, for reasons unrelated to your change, like CI timeouts)

### CODE REVIEW

**Comment by pythonissam** [comment]
ref: 1496

File: yesod-test/Yesod/Test.hs [hunk 1]
```diff
@@ -383,6 +384,12 @@ assertEqual = assertEqualNoShow
 assertEqualNoShow :: (HasCallStack, Eq a) => String -> a -> a -> YesodExample site ()
 assertEqualNoShow msg a b = liftIO $ HUnit.assertBool msg (a == b)
 
+-- | Unconditionally signals that a failure has occured.
+--
+-- @since 1.6.3
```
Comment (snoyberg) [inline yesod-test/Yesod/Test.hs]: Thank you for including the `@since`! Can you also add a `ChangeLog` entry?

Comment (pythonissam) [reply yesod-test/Yesod/Test.hs]: Is this okay??

**Comment by pythonissam** [comment]
Is this okay?

**Comment by snoyberg** [comment]
Sorry, there actually is a problem with the PR, CI is failing:

https://travis-ci.org/yesodweb/yesod/jobs/359647053

**Comment by pythonissam** [comment]
@snoyberg Well, I understand...

**Comment by pythonissam** [comment]
```
test/EmbedProductionTest.hs:41:28: error:
    Ambiguous occurrence ‘assertFailure’
    It could refer to either ‘Test.HUnit.assertFailure’,
                             imported from ‘Test.HUnit’ at test/EmbedProductionTest.hs:11:20-32
                             (and originally defined in ‘Test.HUnit.Lang’)
                          or ‘Yesod.Test.assertFailure’,
                             imported from ‘Yesod.Test’ at test/EmbedProductionTest.hs:15:1-17
```

In this case, which should we modify?  The name of our custom `assertFailure` or the test which uses the original `assertFailure`.

**Comment by snoyberg** [comment]
I think we need to rename this function, in order to avoid a more general baking conflict with a dependency.

**Comment by pythonissam** [comment]
@snoyberg How about this one: `assertFlr`

**Comment by snoyberg** [comment]
I'm not a fan of the abbreviated naming. Let me ask a more basic question: why is `assertFailure` better than just using `fail` or `error`?

**Comment by pythonissam** [comment]
> I'm not a fan of the abbreviated naming

I understand.

> why is assertFailure better than just using fail or error?

Using `assertFailure`, we can obtain more detailed information. For example,

`error` and `fail`:

``````

**Comment by snoyberg** [comment]
That looks like the same information displayed in different formats to me.

**Comment by pythonissam** [comment]
Hmm, come to think of it, you're right... I'm closing this issue.

**Pull request closed without merging**